src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -562,6 +565,10 @@ def cli_execute(self, cmd):
             CONFIG.load(os.path.join(azure_folder, 'az.json'))
             SESSION.load(os.path.join(azure_folder, 'az.sess'), max_age=3600)
 
+            config = Configuration()
+            self.app.initialize(config)
+            self.app.session['headers'] = {'x-ms-client-request-id': str(uuid.uuid1())}
```

Comment:
I don't think this is needed anymore. After this change: https://github.com/Azure/azure-cli/commit/89844da9cd3963c4bcbb89655d948519337af843, the request ID will be refresh when app.execute is invoked.